storage/client_raft_test.go

```diff
@@ -651,3 +652,30 @@ func TestReplicateAddAndRemove(t *testing.T) {
 		verify([]int64{39, 0, 39, 39})
 	}
 }
+
+// TestRaftHeartbeats verifies that coalesced heartbeats are correctly
+// suppressing elections in an idle cluster.
+func TestRaftHeartbeats(t *testing.T) {
+	defer leaktest.AfterTest(t)
+
+	mtc := startMultiTestContext(t, 3)
+	defer mtc.Stop()
+	mtc.replicateRange(1, 0, 1, 2)
+
+	// Capture the initial term and state.
+	status := mtc.stores[0].RaftStatus(1)
+	initialTerm := status.Term
+	if status.SoftState.RaftState != raft.StateLeader {
+		t.Errorf("expected node 0 to initially be leader but was %s", status.SoftState.RaftState)
+	}
+
+	// Wait for several ticks to elapse.
+	time.Sleep(5 * mtc.makeContext().RaftTickInterval)
```

Comment:
> Waiting for N ticks isn't really better thank waiting for N times the tick interval. If we could synthesize a tick, wait for the heartbeat to received, then synthesize the next tick without waiting, then that might be worthwhile, but the number of potential events that might need to be observed in this way would be very large

For this particular case, sure. However:
```
$ git grep -F Sleep\( | wc -l
      35
```

> (Even just considering heartbeats, this test wants to wait for a heartbeat to be received, while other tests want to wait for a heartbeat to not be sent due to coalescing, etc)

I don't understand what you're trying to say here.